changes/16852.md

```diff
@@ -0,0 +1,12 @@
+# PR #16852 Split User_command's stable and regular types
+
+## Summary
+
+Decouple user command's `Stable.Latest.t` and `t` types.
+
+Plus: define `read_all_proofs_from_disk` and `write_all_proofs_to_disk` as `Fn.id` in `zkapp_command.ml`
+and use these definitions in all of the types that depend on `Zkapp_command.t`.
+
+## Impact
+
+Allows transaction's type to contain on-disk-cache refences. A step toward larger RAM optimization effort.
```

Comment:
```suggestion
Allows transaction's type to contain on-disk-cache references. A step toward larger RAM optimization efforts.
```

Reply:
you can probably remove this as @mrmr1993 suggested, since this just uses Fn.id for now in the Zkapp_command module

Reply:
Some user-facing description instead would be good. It's hard to sell to users why they care that we 'split User_command's stable and regular types', but there is value here if described nicely.

Reply:
> Some user-facing description instead would be good. It's hard to sell to users why they care that we 'split User_command's stable and regular types', but there is value here if described nicely.

I struggle to find one beyond what's written. Could you recommend or suggest to drop the changelog altogether?